src/server/game/Scripting/ScriptDefines/UnitScript.cpp

```diff
@@ -61,7 +61,7 @@ uint32 ScriptMgr::DealDamage(Unit* AttackerUnit, Unit* pVictim, uint32 damage, D
         auto const& dmg = script->DealDamage(AttackerUnit, pVictim, damage, damagetype);
         if (dmg != damage)
         {
-            return damage;
+            return dmg;
         }
     }
```

Comment:
```suggestion
        damage = script->DealDamage(AttackerUnit, pVictim, damage, damagetype);
    }
```

Looking through the git history the above looks like the intended behavior

original function: https://github.com/azerothcore/azerothcore-wotlk/blob/b7ee1a9f125824cfe1d2e4c5cb5657bec5abd9db/src/game/Scripting/ScriptMgr.cpp#L1478

with macro expansion

```
uint32 ScriptMgr::DealDamage(Unit* AttackerUnit, Unit *pVictim, uint32 damage, DamageEffectType damagetype)
{
  if (ScriptRegistry<UnitScript>::ScriptPointerList.empty())
    return damage;
  for (ScriptRegistry<UnitScript>::ScriptMapIterator itr =
           ScriptRegistry<UnitScript>::ScriptPointerList.begin();
       itr != ScriptRegistry<UnitScript>::ScriptPointerList.end(); ++itr)
    damage = itr->second->DealDamage(AttackerUnit, pVictim, damage, damagetype);
  return damage;
}
```